src/ast/ast_validations.c

```diff
@@ -1193,7 +1193,10 @@ static VISITOR_STRATEGY _Validate_set_property
 
 	const cypher_astnode_t *ast_prop = cypher_ast_set_property_get_property(n);
 	const cypher_astnode_t *ast_entity = cypher_ast_property_operator_get_expression(ast_prop);
-	if(cypher_astnode_type(ast_entity) != CYPHER_AST_IDENTIFIER) {
+	if(cypher_astnode_type(ast_entity) == CYPHER_AST_PROPERTY_OPERATOR) {
+		ErrorCtx_SetError("Type mismatch: expected Node or Edge but was %s", SIType_ToString(cypher_astnode_type(n)));
```

Comment:
use Error_SITypeMismatch

Reply:
OK, change done. Please validate if it is correct.